### PR TITLE
Update CypherTypes in procedure/function signatures

### PR DESCRIPTION
**Pull request by gem-neo4j**
We are updating the signatures in the output for SHOW FUNCTIONS and SHOW PROCEDURES to match the defaults of the new Cypher Types

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Comment by gem-neo4j** [comment]
@JPryce-Aklundh and @nadja-muller this should be ready for review now the main neo one is ready as well :)

File: modules/ROOT/pages/execution-plans/operators.adoc [hunk 1]
```diff
@@ -4731,7 +4731,7 @@ Batch size 128
 | |               +-----------------------------------+----------------+------+---------+----------------+------------------------+-----------+            |                     |
 | +Sort           | label ASC                         |             10 |    4 |       0 |            536 |                    0/0 |     0.178 | label ASC  | In Pipeline 1       |
 | |               +-----------------------------------+----------------+------+---------+----------------+------------------------+-----------+------------+---------------------+
-| +ProcedureCall  | db.labels() :: (label :: STRING?) |             10 |    4 |         |                |                        |           |            | Fused in Pipeline 0 |
+| +ProcedureCall  | db.labels() :: (label :: STRING) |             10 |    4 |         |                |                        |           |            | Fused in Pipeline 0 |
```
Comment (nadja-muller) [inline modules/ROOT/pages/execution-plans/operators.adoc]: Looks like the alignment is off here.

**Review by nadja-muller — APPROVED** [review]
Looks good. Just a nit.

**Comment by gem-neo4j** [comment]
I realised we should probably let users know about this is the deprecations/additions section, so added that :)

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 2]
```diff
@@ -16,10 +16,41 @@ New features are added to the language continuously, and occasionally, some feat
...
+a|
+label:functionality[]
+label:updated[]
+[source, cypher, role="noheader"]
+----
+SHOW FUNCTIONS YIELD *
+SHOW PROCEDURES YIELD *
+----
+a|
+Updated the `signatures` column in `SHOW FUNCTIONS`, and `SHOW PROCEDURES`.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
Updated the `signatures` column in `SHOW FUNCTIONS` and `SHOW PROCEDURES`.
```

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 3]
```diff
@@ -16,10 +16,41 @@ New features are added to the language continuously, and occasionally, some feat
...
+Procedure and function signatures now conform to the new Cypher Type syntax.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]: Maybe this sentence is not necessary - just go straight to Procedure signatures? 

```suggestion
```

File: modules/ROOT/pages/indexes-for-vector-search.adoc [hunk 4]
```diff
@@ -289,7 +289,7 @@ As a result, the store's space is approximately halved.
 .Signature for `db.create.setVectorProperty` to set a vector property
 [source,syntax,role="noheader",indent=0]
 ----
-db.create.setVectorProperty(node :: NODE?, key :: STRING?, vector :: LIST? OF FLOAT?) :: (node :: NODE?)
+db.create.setVectorProperty(node :: NODE, key :: STRING, vector :: LIST<FLOAT) :: (node :: NODE)
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/indexes-for-vector-search.adoc]:
```suggestion
db.create.setVectorProperty(node :: NODE, key :: STRING, vector :: LIST<FLOAT>) :: (node :: NODE)
```

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 5]
```diff
@@ -16,10 +16,41 @@ New features are added to the language continuously, and occasionally, some feat
...
+
+Procedure signatures now follow the pattern:
+`"procedureName(param1 :: TYPE, param2 :: TYPE, .., paramN :: TYPE) :: (returnParam1 :: TYPE, returnParam2, .., returnParamN :: TYPE)"`
+
+With no return columns:
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
The signature for procedures with no return columns now follows the pattern:
```

Maybe not correct - but in any case I think "With no return columns" needs to have more information.

**Review by JPryce-Aklundh — COMMENTED** [review]
Excellent work @gem-neo4j - I spotted one mistake, and had some questions about the information on the Additions page

**Review by JPryce-Aklundh — APPROVED** [review]

**Pull request merged**